ufo2fontir/src/source.rs

```diff
@@ -626,8 +629,30 @@ impl Work<Context, WorkError> for StaticMetadataWork {
         let glyph_locations = master_locations.values().cloned().collect();
         let glyph_order = glyph_order(default_master, designspace_dir, &self.glyph_names)?;
 
+        // https://unifiedfontobject.org/versions/ufo3/fontinfo.plist/#opentype-os2-table-fields
+        let selection_flags = font_info_at_default
+            .open_type_os2_selection
+            .as_ref()
+            .map(|flags| {
+                flags.iter().fold(SelectionFlags::empty(), |acc, e| {
+                    acc | SelectionFlags::from_bits_truncate(1 << e)
+                })
+            })
+            .unwrap_or_default()
+            | font_info_at_default
```

Comment:
i don't immeditately understand what is going on with this `|` bitwise or operator here after the unwrap_or_default().. What is the intention of joining these two statements with `|` here?

Reply:
We want to match this: https://github.com/googlefonts/ufo2ft/blob/0d2688cd847d003b41104534d16973f72ef26c40/Lib/ufo2ft/outlineCompiler.py#L676-L687

note that whenever you see `getAttrWithFallback` you need to look at the ufo2ft.fontInfoData module and see what is the fallback value (either simple hardcoded default or generated from other attributes) that ufo2ft has.
In the case of `openTypeOS2Selection` it's an empty list `[]`, whereas in the case of `styleMapStyleName`  it has a dynamic fallback that checks for `openTypeNamePreferredSubfamilyName` (which in turn defaults to fontinfo's `styleName` if missing) and checks if the latter is in the RIBBI (Regular, Italic, Bold, Bold Italic) set of names otherwise defaults to "regular".

```python
        # style mapping
        selection = list(getAttrWithFallback(font.info, "openTypeOS2Selection"))
        styleMapStyleName = getAttrWithFallback(font.info, "styleMapStyleName")
        if styleMapStyleName == "regular":
            selection.append(6)
        elif styleMapStyleName == "bold":
            selection.append(5)
        elif styleMapStyleName == "italic":
            selection.append(0)
        elif styleMapStyleName == "bold italic":
            selection += [0, 5]
        os2.fsSelection = intListToNum(selection, 0, 16)
```

Reply:
ah ok I see, you are OR-ing the SelectionFlags from `open_type_os2_selection` (when missing or empty it would be 0 thanks to unwrap_or_default()) with the style flags from style_map_style_name, which is pretty much what ufo2ft is doing. I think you're only missing the fallback treatment for styleMapStyleName

Reply:
Oops, fixed to default to regular and added a comment to make the logic clear, that | is a bit easy to miss